src/domain/vocabulary/test/helper.spec.ts

```diff
@@ -0,0 +1,60 @@
+import type { Deps } from '../command'
+import * as E from 'fp-ts/Either'
+import {
+  GatewayTypeError,
+  LanguageEmptyError,
+  NegativeLimitError,
+  createVocabularyDependenciesWithOptions,
+  withLanguage,
+  withLimit,
+  withVocabularyGateway
+} from '../helper/dependencies'
+import { pipe } from 'fp-ts/lib/function'
+import type { VocabularyPort } from '../port'
+
+type Data = {
+  vocabularyGateway?: VocabularyPort
+  language?: string
+  limit?: number
+  expectedDeps: Deps
+  expectedError?: Error
+}
+
+describe('Considering the createDepsWithOptions() function', () => {
+  describe.each`
+    vocabularyGateway                | language     | limit        | expectedDeps                                                                       | expectedError
+    ${undefined}                     | ${undefined} | ${undefined} | ${Object.create(null)}                                                             | ${undefined}
```

Comment:
According to the type, `Deps` cannot be empty. All the properties must exist and be valuated.

Reply:
Yep, will be modified accordingly